src/v2/api/services/RoleService.spec.ts

```diff
@@ -0,0 +1,154 @@
+import { RoleService } from './RoleService';
+import { Test } from '@nestjs/testing';
+import { PrismaModule } from '../../modules/PrismaModule';
+import { PermissionService } from './PermissionService';
+import { RoleRepository } from '../../database/repositories/RoleRepository';
+import { DbRole } from '../../database/entities/DbRole';
+import { CreateGrantDTO } from '../dtos/CreateGrantsDTO';
+
+describe('RoleService', () => {
+  let roleService: RoleService;
+  let permissionService: PermissionService;
+  let roleRepository: RoleRepository;
+
+  beforeEach(async () => {
+    const testingModule = await Test.createTestingModule({
+      providers: [RoleService, PermissionService],
+      imports: [PrismaModule],
+    }).compile();
+
+    roleService = testingModule.get(RoleService);
+    permissionService = testingModule.get(PermissionService);
+    roleRepository = testingModule.get(RoleRepository);
+  });
+
+  describe('canCreateGrants', () => {
+    it('should true if user has grants that is in grants array', async () => {
+      const userRoles = [{
+        grants: [{
+          permission: 'one',
+          set: true,
+        }],
+      }, {
+        grants: [{
+          permission: 'two',
+          set: true,
+        }, {
+          permission: 'three',
+          set: true,
+        }],
+      }] as DbRole[];
```

Comment:
Unfortunately, I can't use declaration of type, as you suggest, because objects, that i create, contain not all fields that are necessary for this type.